app/api/event_copyright.py

```diff
@@ -37,37 +38,50 @@ class Meta:
 
 
 class EventCopyrightList(ResourceList):
-
-    def query(self, view_kwargs):
-        query_ = self.session.query(EventCopyright)
-        if view_kwargs.get('id'):
-            query_ = query_.join(Event).filter(Event.id == view_kwargs['id'])
-        elif view_kwargs.get('identifier'):
-            query_ = query_.join(Event).filter(Event.identifier == view_kwargs['identifier'])
-        return query_
-
     def before_create_object(self, data, view_kwargs):
-        if view_kwargs.get('id'):
-            event = self.session.query(Event).filter_by(id=view_kwargs['id']).one()
-            data['event_id'] = event.id
-        elif view_kwargs.get('identifier'):
-            event = self.session.query(Event).filter_by(identifier=view_kwargs['identifier']).one()
+
+        # Permission Manager will ensure that event_id is fetched into
+        # view_kwargs from event_identifier
+        try:
+            event = self.session.query(Event).filter_by(id=view_kwargs['event_id']).one()
+        except NoResultFound:
+            raise ObjectNotFound({'parameter': 'event_identifier'},
+                                 'Event: {} not found'.format(view_kwargs['event_id']))
```

Comment:
Again, If you pull the latest nextgen branch, you will see it there is any `event_identifier` in url then the permission manager takes cares of it.

Also, for the error. Since it is ensured that `event_id` is there so I directly used it there. Other possible option could be directly saying `Specified Event not found`